adafruit_displayio_layout/widgets/icon_animated.py

```diff
@@ -0,0 +1,243 @@
+from adafruit_displayio_layout.widgets.easing import quadratic_easeout as easein
+from adafruit_displayio_layout.widgets.easing import quadratic_easein as easeout
+
+
+class IconAnimated(IconWidget):
+
+    """
+    An animated touch enabled widget that holds an icon image loaded with
+    OnDiskBitmap and a text label centered beneath it. Includes optional
+    animation to increase the icon size when pressed.
+
+    .. Warning::  The `init_class` class function should be called before instancing any
+        IconAnimated widgets.
+
+    :param str label_text: the text that will be shown beneath the icon image.
+    :param str icon: the filepath of the bmp image to be used as the icon.
+    :param bool on_disk: if True use OnDiskBitmap instead of imageload.
+     This can be helpful to save memory. Defaults to False
+
+    :param float max_scale: the maximum zoom during animation, set 1.0 for no animation
+     a value of 1.4 is a good starting point (default: 1.0, no animation),
+     ``max_scale`` must be between 1.0 and 1.5.
+    :param float max_angle: the maximum degrees of rotation during animation, positive values
+     are clockwise, set 0 for no rotation, in degrees (default: 0 degrees)
+    :param float animation_time: the time for the animation in seconds, set to 0.0 for
+     no animation, a value of 0.15 is a good starting point (default: 0.0 seconds)
+
+    :param int x: x location the icon widget should be placed. Pixel coordinates.
+    :param int y: y location the icon widget should be placed. Pixel coordinates.
+    :param anchor_point: (X,Y) values from 0.0 to 1.0 to define the anchor point relative to the
+     widget bounding box
+    :type anchor_point: Tuple[float,float]
+    :param int anchored_position: (x,y) pixel value for the location of the anchor_point
+    :type anchored_position: Tuple[int, int]
+    :param int max_size: (Optional) this will get passed through to the
+     displayio.Group constructor. If omitted we default to
+     grid_size width * grid_size height to make room for all (1, 1) sized cells.
+    """
+
+    # pylint: disable=bad-super-call, too-many-instance-attributes, too-many-locals
+    # pylint: disable=too-many-arguments, unused-argument
+
+    display = None
+    # The other Class variables are created in Class method `init_class`:
+    #               max_scale, bitmap_buffer, palette_buffer
+
+    @classmethod
+    def init_class(
+        cls, display=None, max_scale=1.5, max_size=(80, 80), max_color_depth=512
+    ):
+        """
+        Initializes the IconAnimated Class variables, including preallocating memory
+        buffers for the icon zoom bitmap and icon palette.
+
+        .. Note::  The `init_class` class function should be called before instancing any
+                IconAnimated widgets. Usage example:
+                ``IconAnimated(display=board.DISPLAY, max_scale=1.5,
+                max_size=(80,80), max_color_depth=256)``
```

Comment:
Ah, I figured out the proper syntax. 

I think this usage example needs the `init_class()` added to it.
```
IconAnimated.init_class(display=board.DISPLAY, max_scale=1.5, max_size=(80,80), max_color_depth=256)
```

Reply:
Yikes, sorry for the error in the docs, that must have been really confusing.  I appreciate that you caught this since that's the important part!